zilencer/views.py

```diff
@@ -218,16 +230,17 @@ def register_remote_server(
     return json_success(request, data={"created": created})
 
 
-@has_request_variables
+@typed_endpoint
 def register_remote_push_device(
     request: HttpRequest,
     server: RemoteZulipServer,
-    user_id: Optional[int] = REQ(json_validator=check_int, default=None),
-    user_uuid: Optional[str] = REQ(default=None),
-    realm_uuid: Optional[str] = REQ(default=None),
-    token: str = REQ(),
-    token_kind: int = REQ(json_validator=check_int),
-    ios_app_id: Optional[str] = REQ(str_validator=check_app_id, default=None),
+    *,
+    user_id: Optional[Json[int]] = None,
+    user_uuid: Optional[str] = None,
+    realm_uuid: Optional[str] = None,
+    token: Annotated[str, RequiredStringConstraint],
+    token_kind: Json[int],
+    ios_app_id: Annotated[Optional[str], StringConstraints(pattern="[.a-zA-Z0-9-]+")] = None,
```

Comment:
`StringConstraints(pattern="[.a-zA-Z0-9-]+")`, should check for a valid regex pattern but it's falling to do So. Investigating

Reply:
reason behind pattern not being full matched,  
https://docs.pydantic.dev/2.0/migration/#patterns-regex-on-strings